### PR TITLE
Jonker/tabs

### PR DESCRIPTION
**Pull request by bryanjonker-illinois**
Moved code to connectedCallback and render. 
Removed anonymous functions
Fixed arrow keys

### CODE REVIEW

File: src/components/il-tabs/il-tabs.component.js [hunk 1]
```diff
@@ -94,6 +117,7 @@ class TabComponent extends LitElement {
   }
 
   render() {
+    this.setTabs();
     return html`
         <div id="container" ?compact=${this.compact}>
```
Comment (mattsharkey) [inline src/components/il-tabs/il-tabs.component.js:122]: The compact attribute is just for the polyfill right? I think it should be a CSS class

File: src/components/il-tabs/il-tabs.component.js [hunk 2]
```diff
@@ -94,6 +117,7 @@ class TabComponent extends LitElement {
   }
 
   render() {
+    this.setTabs();
```
Comment (mattsharkey) [inline src/components/il-tabs/il-tabs.component.js:120]: I'm not sure we want to do this on every render. Under what circumstances do we *need* to reformat the tabs?

File: src/components/il-tabs/il-tabs.component.js [hunk 3]
```diff
@@ -25,42 +25,62 @@ class TabComponent extends LitElement {
...
       item.setAttribute('role', 'tab');
-      if (firstItem) {
+      if (i == 0) {
         item.setAttribute('tabindex', 0);
         item.setAttribute('aria-selected', 'true');
         this.setActivePanel(item);
-        firstItem = false;
       } else {
         item.setAttribute('tabindex', -1);
       }
```
Comment (mattsharkey) [inline src/components/il-tabs/il-tabs.component.js:78]: Can these be a separate class method, initTab() or something more descriptive of what it's doing?

File: src/components/il-tabs/il-tabs.component.js [hunk 4]
```diff
@@ -25,42 +25,62 @@ class TabComponent extends LitElement {
     this.compact = this.offsetWidth < TabComponent.compactSizePixelWidth;
   }
 
-   getLinkElement() {
-     return document.getElementById(this.linkedby);
+   addPolyfillForContainerSupprt() {
```
Comment (mattsharkey) [inline src/components/il-tabs/il-tabs.component.js:28]: It's not really important that you're polyfilling something. The method name should describe what you're doing, e.g. addResizeListeners()

File: src/components/il-tabs/il-tabs.component.js [hunk 5]
```diff
@@ -75,14 +95,17 @@ class TabComponent extends LitElement {
     item.setAttribute('aria-selected', 'true');
   }
 
-  keyboard(event) {
-    if (event.code == "ArrowLeft" || event.code == "ArrowUp") {
+  handleKeypress(event) {
+    // if vertical (compact or il-vertical-tabs), use up and down arrows -- otherwise, use left and right arrows
+    let tabComponent = event.target.closest('il-tabs');
+    var arrowDirectionVertical = tabComponent.compact || tabComponent.classList.contains('il-vertical-tabs');
```
Comment (mattsharkey) [inline src/components/il-tabs/il-tabs.component.js:101]: Can this be a separate method, e.g. isVertical() or something?

**Review by mattsharkey — CHANGES_REQUESTED** [review]

File: src/components/il-tabs/il-tabs.component.js (hunk 2 above)
Comment (bryanjonker-illinois) [reply src/components/il-tabs/il-tabs.component.js:129]: I think we do need to do this on every render to manage the case where things get added or removed.

Comment (mattsharkey) [reply src/components/il-tabs/il-tabs.component.js:129]: Did you try the mutation events for that?

Comment (bryanjonker-illinois) [reply src/components/il-tabs/il-tabs.component.js:129]: I tried, but I may have been doing it wrong. I created an issue for this

**Pull request merged**